### DISCUSSION TITLE
Linking auth providers to an existing account when signups are disabled

### DISCUSSION THREAD

**Discussion by Groszczu** [supabase/supabase#11800] [Questions] [answered]
I'm building an app where only invited users can join , therefore I disabled the `Allow new users to sign up` option in a Supabase > Settings > Auth panel. 

![image](https://user-images.githubusercontent.com/45833713/213702246-91ed0ce9-f5f7-43b1-9203-bae876b0ab8d.png)

The issue is that users who accepted an email invite are not able to sign in using other auth providers (for example Google) when this option is disabled.
Not sure if it is a bug or if is it intentional. Linking auth provider to an existing account is not exactly a sign-up (no new user is created in the database) so I guess it should be allowed.
I would like existing users to link other auth providers to their accounts and prevent new users from signing up without an invite.
Is it possible with a Supabase config or do I have to implement my own invite-only sign-up flow?

**Comment by hf** [comment, marked as answer]
> Not sure if it is a bug or if is it intentional. Linking auth provider to an existing account is not exactly a sign-up (no new user is created in the database) so I guess it should be allowed.

It is intentional. Sign up means creating an account when one did not exist before, regardless of the provider.

> I would like existing users to link other auth providers to their accounts and prevent new users from signing up without an invite.
Is it possible with a Supabase config or do I have to implement my own invite-only sign-up flow?

Please emote on the [Anonymous Sign-in](https://github.com/supabase/gotrue/issues/68) issue to be notified on progress on this.
